### PR TITLE
Try existing installation when auto or binary

### PR DESCRIPTION
**Pull request by meatballhat**
plus docs and -f/--force

### CODE REVIEW

File: gimme [hunk 1]
```diff
@@ -316,13 +364,16 @@ unset GOOS
 unset GOPATH
 unset GOROOT
 
+GIMME_VERSION_PREFIX="$(_realpath "${GIMME_VERSION_PREFIX}")"
+GIMME_ENV_PREFIX="$(_realpath "${GIMME_ENV_PREFIX}")"
```
Comment (tianon) [inline gimme:370]: I'm not sure I understand why `_realpath` is necessary here?

Comment (meatballhat) [inline gimme:370]: It's a minor nitpick, but I found this to be non-awesome:

``` bash
gimme 1.4 .
```

results in an env file with the following:

```
export GOROOT="./go1.4.darwin.amd64"
```

Comment (meatballhat) [inline gimme:370]: And I'm using a func version instead of `readlink -f` because of the great GNU v. BSD schism.

Comment (tianon) [inline gimme:370]: Ahhh.  Ok.  I think I can improve on `_realpath` slightly then (although OSX support is tricky here for a clean solution -- ideally we'd just `readlink -f` but `greadlink` isn't common on OSX :skull:).

Comment (meatballhat) [inline gimme:370]: Yes please improve :smiley_cat: :exclamation:

File: gimme [hunk 2]
```diff
@@ -256,6 +277,28 @@ _try_git() {
...
+
+	if [[ -s "${env_file}" ]] ; then
+		rm -rf "$(awk -F\" '/GOROOT/ { print $2 }' "${env_file}")"
+		rm -f "${env_file}"
+	fi
+}
+
+_realpath() {
+	[[ $1 = /* ]] && echo "$1" || echo "$PWD/${1#./}"
+}
```
Comment (tianon) [inline gimme:291]: So, this feels a little bit hacky, and I think something like the following will be somewhat more robust:

``` console
[ -d "$1" ] && echo "$(cd "$1" && pwd)" || echo "$(cd "$(dirname "$1")" && pwd)/$(basename "$1")"
```

ie:

``` console
$ cat temp.sh
#!/bin/bash
set -e

_realpath() {
    echo -n "$1 -->> "
    [ -d "$1" ] && echo "$(cd "$1" && pwd)" || echo "$(cd "$(dirname "$1")" && pwd)/$(basename "$1")"
}

_realpath .
_realpath ..
_realpath /tmp/..
_realpath /home/../home/tianon/
_realpath "$BASH_SOURCE"
$ bash temp.sh
. -->> /home/tianon
.. -->> /home
/tmp/.. -->> /
/home/../home/tianon/ -->> /home/tianon
temp.sh -->> /home/tianon/temp.sh
```

Comment (tianon) [inline gimme:291]: ... and if we want almost fully canonical paths (ala `readlink -f`), we could swap those `pwd` invocations for `pwd -P`, but IMO that's overkill here :+1:

Comment (meatballhat) [inline gimme:291]: :clap: :smiley:

**Comment by tianon** [comment]
LGTM

**Pull request merged**